### ISSUE TITLE
Config watcher didn't response to configmap change after 45 minutes

### ISSUE THREAD

**Issue by neo-feng** [spring-cloud/spring-cloud-kubernetes#837]
**Describe the bug**
I'm using config watcher to enable auto refresh for my  spring cloud microservices deployed in aws EKS.
Config watcher is also deployed in EKS. 
It works fine when config watcher was just deployed, however, after 40 minutes, when I modified the configmap again to test it again, config watcher didn't response, no logs at all.

Config watcher version:  2.1.0-SNAPSHOT
Spring boot version: 2.4.4
JDK: 1.8.0_282

**Sample**
project link:  https://github.com/neo-feng/configdemo 
you may find all information from link above.

pom dependencies:
```
	<dependencies>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-web</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-actuator</artifactId>
		</dependency>
		<dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-starter-kubernetes-fabric8-config</artifactId>
            <version>${spring.cloud.config.version}</version>
        </dependency>
	</dependencies>
```

bootstrap.yml
```
spring:
  application:
    name: demo-service
  cloud:
    kubernetes:
      config:
        #enabled: true
        sources:
          - namespace: default
            name: ${spring.application.name}

server:
  port: 8080
```
reproduce steps:
- deploy the application and test it successfully according to readme.md in the link above
- after 40 minutes, test it again -- change the configmap and watch logs
- you'll see config watcher has no response, no logs at all

It seems config watcher was disconnected with kubernetes after 40 minutes, while it still works well after 20 minutes, has no idea how to proceed, any suggestion will be appreciated.

**Comment by trettstadtnlb** [comment]
I have exactly the same problem when the configuration watcher (version 2.0.3) is deployed in AKS while it works with a locally deployed version (Docker Desktop, Windows 10). Even with the log level set to DEBUG, there are no logs after some time and the applications are no longer notified.
@neo-feng Your project repository seems to be private, because I get a 404 when clicking the link.

**Comment by neo-feng** [comment]
> I have exactly the same problem when the configuration watcher (version 2.0.3) is deployed in AKS while it works with a locally deployed version (Docker Desktop, Windows 10). Even with the log level set to DEBUG, there are no logs after some time and the applications are no longer notified.
> @neo-feng Your project repository seems to be private, because I get a 404 when clicking the link.

Just changed the visibility of demo project, please try again. @trettstadtnlb

**Comment by trettstadtnlb** [comment]
@neo-feng I can see the repository now.

**Comment by arobie1992** [comment]
This sounds almost identical to the issue I'm running into. Had a ticket here, though the name is inaccurate due to an original misdiagnosis of the cause.

https://github.com/spring-cloud/spring-cloud-kubernetes/issues/831

Let me know if you have any luck. I'll let you know if I find anything.

**Comment by ryanjbaxter** [comment]
Fixed via 888031d
